lib/pp.ml

```diff
@@ -297,3 +311,66 @@ let prvect_with_sep sep elem v = prvecti_with_sep sep (fun _ -> elem) v
 let prvect elem v = prvect_with_sep mt elem v
 
 let surround p = hov 1 (str"(" ++ p ++ str")")
+
+(*** DEBUG code ***)
+
+let db_print_pp ?(fmt=Format.std_formatter) pp =
+  let open Format in
+  let block_type btype =
+    let (bt, v) =
+      match btype with
+      | Pp_hbox v -> ("Pp_hbox", v)
+      | Pp_vbox v -> ("Pp_vbox", v)
+      | Pp_hvbox v -> ("Pp_hvbox", v)
+      | Pp_hovbox v -> ("Pp_hovbox", v)
+    in
+    Printf.sprintf "%s %d" bt v
+  in
+  let rec db_string_of_pp_r pp indent =
+    let ind () = fprintf fmt "%s" (String.make (2 * indent) ' ') in
+    ind();
+    match pp with
+    | Ppcmd_empty ->
+        fprintf fmt "Ppcmd_empty\n"
+    | Ppcmd_string str ->
+        fprintf fmt "Ppcmd_string '%s'\n" str
+    | Ppcmd_glue list ->
+        fprintf fmt "Ppcmd_glue (\n";
+        List.iter (fun x -> db_string_of_pp_r (repr x) (indent + 1)) list; ind ();
+        fprintf fmt ") Ppcmd_glue\n"
+    | Ppcmd_box (block, pp) ->
+        fprintf fmt "Ppcmd_box %s\n" (block_type block);
+        (fun x -> db_string_of_pp_r (repr x) (indent + 1)) pp
```

Comment:
The fun is unnecessary, also this should be done in a single fprintf call using the `%a` modifier.